vscode/vscode/vscode.ml

```diff
@@ -1810,39 +1812,254 @@ module Workspace = struct
     [@@js.global "vscode.workspace.openTextDocument"]
 end
 
+module TreeItemCollapsibleState = struct
+  type t =
+    | None [@js 0]
+    | Collapsed [@js 1]
+    | Expanded [@js 2]
+  [@@js.enum] [@@js]
+end
+
+module TreeItemLabel = struct
+  type t = private (* interface *) Ojs.t [@@js]
+
+  val label : t -> string [@@js.get]
+
+  val highlights : t -> (int * int) list or_undefined [@@js.get]
+
+  val create : label:string -> ?highlights:(int * int) list -> unit -> t
+    [@@js.builder]
+end
+
+module ThemeIcon = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  val make : id:string -> ?color:ThemeColor.t -> unit -> t
+    [@@js.new "vscode.ThemeIcon"]
+
+  val file : t [@@js.global "vscode.ThemeIcon.File"]
+
+  val folder : t [@@js.global "vscode.ThemeIcon.Folder"]
+
+  val id : t -> string [@@js.get]
+
+  val color : t -> ThemeColor.t option [@@js.get]
+end
+
+module TreeItem = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  type lightDarkIcon =
+    { light : ([ `String of string | `Uri of Uri.t ][@js.union])
+    ; dark : ([ `String of string | `Uri of Uri.t ][@js.union])
+    }
+  [@@js]
+
+  type iconPath =
+    ([ `String of string
+     | `Uri of Uri.t
+     | `LightDark of lightDarkIcon
+     | `ThemeIcon of ThemeIcon.t
+     ]
+    [@js.union])
+  [@@js]
+
+  let iconPath_of_js js_val =
+    if Ojs.has_property js_val "path" then
+      `Uri ([%js.to: Uri.t] js_val)
+    else if Ojs.has_property js_val "id" then
+      `ThemeIcon ([%js.to: ThemeIcon.t] js_val)
+    else if Ojs.type_of js_val = "string" then
+      `String ([%js.to: string] js_val)
+    else
+      assert false
```

Comment:
For `iconPath_of_js`, what if js_val is a lightDarkIcon? That would assert false right now.

Reply:
Done